spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestMetadataTablesWithPartitionEvolution.java

```diff
@@ -261,6 +262,88 @@ public void testEntriesMetadataTable() throws ParseException {
     }
   }
 
+  @Test
+  public void testPartitionMetadataTable() throws ParseException {
```

Comment:
As one more thing to test, can you check reordering partition transforms? 
Going from data, category -> category,data

I think you have this covered but I want to make sure we have a test in there since I think this is a pretty common usecase

Reply:
Added the test.  Unfortunately it's a bit weird until https://github.com/apache/iceberg/issues/4292. completely fixed (I believe @szlta is taking a look at the remaining point there which would fix this issue).